lib/natalie/compiler/pass0c.rb

```diff
@@ -98,11 +124,19 @@ def process_call(exp, block_with_args = nil)
         _, receiver, method, *args = exp
         if receiver && !receiver.empty?
           t, *values = receiver
-          if t == :array
-            if method == :min && args.empty? && !block_with_args
-              return process(canary_array_min(exp, values))
-            elsif method == :max && args.empty? && !block_with_args
-              return process(canary_array_max(exp, values))
+          if t == :array && block_with_args.nil?
+            # FIXME: Maybe throw argument errors when we detect anything wrong with the argument count
+            case method
+            when :include?
+              return process(canary_array_include(exp, values, *args)) if args.length == 1
+            when :min
+              return process(canary_array_min(exp, values)) if args.length == 0
+            when :max
+              return process(canary_array_max(exp, values)) if args.length == 0
+            when :any?
+              return process(canary_array_any(exp, values)) if args.length == 0
+            when :all?
+              return process(canary_array_all(exp, values)) if args.length == 0
```

Comment:
I believe the `canary_array_whatever` methods should decide which exps to process... otherwise I have to jump back and forth a couple places to know if an optimization is running...

Reply:
I see that point,
but with the current architecture this would make it loop processing the same statement,
because we would try to reprocess the if that it should spit out, but it's still the call it was

In this case these checks should be an ArgumentErrors, I guess...
I will move them into the functions for now, but going forward separating this into multiple optimization passes would be the best way going forward, as discussed in the discord